lib/Sema/CSGen.cpp

```diff
@@ -1724,6 +1724,22 @@ namespace {
 
     Type visitTypeValueExpr(TypeValueExpr *E) {
       auto ty = E->getParamDecl()->getDeclaredInterfaceType();
```

Comment:
It looks like a TypeValueExpr either represents an unqualified reference `n` or a member reference `Foo<...>.n`. The paramDecl is always set, and the latter also sets a TypeRepr. You can model both cases with a single TypeRepr instead